src/main/java/echo/parser/Parser.java

```diff
@@ -0,0 +1,280 @@
+package echo.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import echo.command.ByeCommand;
+import echo.command.Command;
+import echo.command.DeadlineCommand;
+import echo.command.DeleteCommand;
+import echo.command.EventCommand;
+import echo.command.FindCommand;
+import echo.command.HelpCommand;
+import echo.command.ListCommand;
+import echo.command.MarkCommand;
+import echo.command.TodoCommand;
+import echo.command.UnmarkCommand;
+import echo.utils.EchoException;
+
+/**
+ * Parser encapsulates a parser to deal with making sense of the user command.
+ */
+public class Parser {
+
+    /** Is user exited */
+    private static boolean isExit = false;
+
+    /**
+     * Parses user command and determines action to perform.
+     *
+     * @param input Input from user.
+     * @return Command to be executed.
+     *
+     * @throws EchoException If input is invalid.
+     */
+    public static Command parse(String input) throws EchoException {
+        try {
+            String[] splitSpace = input.split(" ");
+            String command = splitSpace[0];
+
+            switch (command) {
+            case ListCommand.COMMAND:
+                return prepareListCommand();
+            case TodoCommand.COMMAND:
```

Comment:
Indentation levels for the switch commands are correct! Consider adding a line between each case to enhance readability! :)